flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/operator/AbstractKeyedCEPPatternOperator.java

```diff
@@ -126,6 +129,7 @@ public AbstractKeyedCEPPatternOperator(
 		this.isProcessingTime = Preconditions.checkNotNull(isProcessingTime);
 		this.keySerializer = Preconditions.checkNotNull(keySerializer);
 		this.nfaFactory = Preconditions.checkNotNull(nfaFactory);
+		this.conditionRegistry = nfaFactory.createNFA().getConditionRegistry();
```

Comment:
I have tried that, but found it would touch too many codes.  We need to change `process` signature to add the `conditionRegistry` parameter, this need to change many codes. And we use `NFACompiler.compile` to compile the given pattern into a  NFA, but without `conditionRegistry`, that's why I have to put `conditionRegistry` in NFA.

This may look a bit wired. But I'm not sure which way is better.